Dockerfile

```
@@ -71,44 +54,21 @@ RUN \
   if test -n "$CABAL_VERSION"; then \
     ghcup install cabal "$CABAL_VERSION" --set; \
     cabal --version; \
-  fi
-
-# Configure Cabal.
-
-ARG CABAL_STORE=/cabal-store
-RUN \
-  set -o errexit -o xtrace; \
-  if command -v cabal; then \
-    sudo mkdir --mode 0775 --parents --verbose "$CABAL_STORE"; \
-    sudo chown --verbose "$USER_NAME" "$CABAL_STORE"; \
-    sudo chgrp --verbose sudo "$CABAL_STORE"; \
-    cabal user-config init --augment "store-dir: $CABAL_STORE"; \
+    cabal user-config init; \
   fi
 
 # Install Stack.
 
 ARG STACK_VERSION=2.9.3
+ENV STACK_XDG=1
 RUN \
   set -o errexit -o xtrace; \
   if test -n "$STACK_VERSION"; then \
     ghcup install stack "$STACK_VERSION" --set; \
     stack --version; \
+    stack config set system-ghc --global false; \
```

In fact, we should also avoid installing GHC.
```suggestion
    stack config set system-ghc --global false; \
    stack config set install-ghc --global false; \
```